tests/unit/util/test_crs.py

```diff
@@ -0,0 +1,102 @@
+            standard_parallels=(64.25, 65.75),
+            globe=ccrs.Globe(ellipse="GRS80"),
+        )
+    ),
+
+    (
+        # Not sure how else to define this case other than copying what
+        # our cartopy -> pyproj converter does...
+        pyproj.CRS.from_epsg(
+            """
+            +ellps=WGS84 +a=6371229 +b=6371229
+            +proj=ob_tran +o_proj=latlon +o_lon_p=0.0 +o_lat_p=177.5
+            +lon_0=217.5
+            +to_meter=0.0174532925199433 +no_defs
+            """
+        ),
+        ccrs.RotatedGeodetic(
+            37.5, 177.5,
+            globe=ccrs.Globe(semimajor_axis=6371229, semiminor_axis=6371229),
+        )
+    ),
+]
+
+@pytest.mark.parametrize("source, target", cases)
+def test_as_cartopy(source, target):
+    converted = util.crs.as_cartopy_crs(source)
+    # Cartopy implements equality by comparing proj4 strings, which may
+    # fail due to differences such as the order of parameters, or whether a
+    # number ends with ".0" or not.  Instead, compare the underlying dicts.
+    assert converted.proj4_params == target.proj4_params
+
+# Note that we can reuse the same cases, but labeled the other way round
+@pytest.mark.parametrize("target, source", cases)
+def test_as_pyproj(source, target):
+    converted = util.crs.as_pyproj_crs(source)
+    # Check only proj4 parameters.  Pyproj can handle a lot more than proj4
+    # can describe, so may have got some of those extra things wrong.
+    assert converted.to_dict() == target.to_dict()
+
+
+class TestTransform():
+    def setup_class(self):
+        self.latlon_point = shapely.geometry.Point(-0.4, 51.5)
+        self.osgb_point = shapely.geometry.Point(511160, 179110)
+
+    def test_cartopy(self):
+        latlon = ccrs.Geodetic()
+        osgb = ccrs.OSGB(approx=False)
+        transformed = util.crs.transform_shape(self.latlon_point, latlon, osgb)
+        transformed = np.array(transformed).round(-1)
+        assert np.all(transformed == np.array(self.osgb_point))
+
+    def test_pyproj(self):
+        latlon = pyproj.CRS.from_epsg(4326)
+        osgb = pyproj.CRS.from_epsg(27700)
+        transformed = util.crs.transform_shape(self.latlon_point, latlon, osgb)
+        transformed = np.array(transformed).round(-1)
+        assert np.all(transformed == np.array(self.osgb_point))
+
+    def test_cartopy_to_pyproj(self):
+        latlon = ccrs.Geodetic()
+        osgb = pyproj.CRS.from_epsg(27700)
+        transformed = util.crs.transform_shape(self.latlon_point, latlon, osgb)
+        transformed = np.array(transformed).round(-1)
+        assert np.all(transformed == np.array(self.osgb_point))
+
+    def test_pyproj_to_cartopy(self):
+        latlon = pyproj.CRS.from_epsg(4326)
+        osgb = ccrs.OSGB(approx=False)
+        transformed = util.crs.transform_shape(self.latlon_point, latlon, osgb)
+        transformed = np.array(transformed).round(-1)
+        assert np.all(transformed == np.array(self.osgb_point))
```

Comment:
This is all really lovely.  I think I must be on the scale a little bit because this layout is really doing it for me.  Very pretty.  And of course effective.